refactor(core): move facades out of core

vsavkin opened the pull request:
This is part of ongoing work to make core platform-independent.

BREAKING CHANGE

All private exports from 'angular2/src/core/facade/{lang,collection,exception_handler}' should be replaced with 'angular2/src/facade/{lang,collection,exception_handler}'.

mary-poppins commented:
Merging PR #5167 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5167](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5167).

vsavkin committed:
chore(ts2dart): update ts2dart to 0.7.14

mary-poppins commented:
Merging PR #5167 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5167](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5167).

vsavkin committed:
refactor(core): move facades out of core

This is part of ongoing work to make core platform-independent.

BREAKING CHANGE

All private exports from 'angular2/src/core/facade/{lang,collection,exception_handler}' should be replaced with 'angular2/src/facade/{lang,collection,exception_handler}'.

mary-poppins commented:
Merging PR #5167 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5167](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5167).
